src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/index.ts

```diff
@@ -181,3 +146,83 @@ export const thorchainSaversStakingOpportunitiesMetadataResolver = async ({
 
   return { data }
 }
+
+export const thorchainSaversStakingOpportunitiesUserDataResolver = async ({
+  opportunityType,
+  accountId,
+  reduxApi,
+}: OpportunitiesUserDataResolverInput): Promise<{ data: GetOpportunityUserStakingDataOutput }> => {
+  const { getState } = reduxApi
+  const state: any = getState() // ReduxState causes circular dependency
+
+  const stakingOpportunitiesUserDataByUserStakingId: OpportunitiesState['userStaking']['byId'] = {}
+
+  const stakingOpportunityId = accountIdToFeeAssetId(accountId)
+
+  if (!stakingOpportunityId)
+    throw new Error(`Cannot get stakingOpportunityId for accountId: ${accountId}`)
+
+  const asset = selectAssetById(state, stakingOpportunityId)
+  if (!asset) throw new Error(`Cannot get asset for stakingOpportunityId: ${stakingOpportunityId}`)
+
+  const toAssetIdParts: ToAssetIdArgs = {
+    assetNamespace: fromAssetId(stakingOpportunityId).assetNamespace,
+    assetReference: fromAssetId(stakingOpportunityId).assetReference,
+    chainId: fromAssetId(stakingOpportunityId).chainId,
+  }
+  const opportunityId = toOpportunityId(toAssetIdParts)
+  const userStakingId = serializeUserStakingId(accountId, opportunityId)
+
+  const allPositions = await getThorchainSaversPositions(stakingOpportunityId)
+
+  if (!allPositions.length)
+    throw new Error(
+      `Error fetching THORCHain savers positions for assetId: ${stakingOpportunityId}`,
+    )
+
+  // Returns either
+  // - A tuple made of a single address for EVM and Cosmos chains since the address *is* the account
+  // - An array of many addresses for UTXOs, since an xpub can derive many many addresses
+  const accountAddresses = await getAccountAddresses(accountId)
+
+  // TODO(gomes): This is wrong for UTXOs. We need to pass an address, not an AccountId
```

Comment:
this comment still valid?

Reply:
https://github.com/shapeshift/web/pull/3613/commits/8d4cd8733e951de3fabc0c7871eb890f9f510a43